libctl/structure.cpp

```diff
@@ -1300,8 +1297,18 @@ static meep::susceptibility *make_multilevel_sus(const multilevel_atom *d) {
   for (int t = 0; t < d->transitions.num_items; ++t) {
     int i = d->transitions.items[t].from_level - minlev;
     int j = d->transitions.items[t].to_level - minlev;
-    Gamma[i*L+i] += d->transitions.items[t].transition_rate;
-    Gamma[j*L+i] -= d->transitions.items[t].transition_rate;
+    Gamma[i*L+j+1] -= d->transitions.items[t].transition_rate;
+    Gamma[(i-1)*L+j+1] += d->transitions.items[t].transition_rate;
+  }
+  
+  // pumping rate matrix Rp
+  meep::realnum *Rp = new meep::realnum[L * L];
+  memset(Rp, 0, sizeof(meep::realnum) * (L*L));
+  for (int t = 0; t < d->transitions.num_items; ++t) {
+    int i = d->transitions.items[t].from_level - minlev;
+    int j = d->transitions.items[t].to_level - minlev;
+    Rp[i*L+i] -= d->transitions.items[t].pumping_rate;
```

Comment:
No, I guess `i*L+i` is correct here.  This should be the same as Γ, above (but see the discussion above about your changes to the Γ indexing).